instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/ConsumerRecordsInstrumentation.java

```diff
@@ -65,14 +63,13 @@ public static class IterableAdvice {
     public static <K, V> void wrap(
         @Advice.This ConsumerRecords<?, ?> records,
         @Advice.Return(readOnly = false) Iterable<ConsumerRecord<K, V>> iterable) {
-      // typically, span key suppression should happen inside the Instrumenter, but receiveContext
-      // is being used as the parent context for the span instead of the current context
-      if (iterable != null
-          && SpanKey.CONSUMER_PROCESS.fromContextOrNull(currentContext()) == null) {
-        Context receiveContext =
-            VirtualField.find(ConsumerRecords.class, Context.class).get(records);
-        iterable = TracingIterable.wrap(iterable, receiveContext);
-      }
+
+      // it's important not to suppress consumer span creation here because this instrumentation can
+      // leak the context and so there may be a leaked consumer span in the context, in which
+      // case it's important to overwrite the leaked span instead of suppressing the correct span
+      // (https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1947)
+      Context receiveContext = VirtualField.find(ConsumerRecords.class, Context.class).get(records);
+      iterable = TracingIterable.wrap(iterable, receiveContext);
```

Comment:
It looks like this is breaking the spring-kafka instrumentation: it expects that its batch consumer process span will suppress the kafka-clients individual record consumer process spans.
Hm, should we add a thread local "don't create consumer process spans" for kafka-clients?

Reply:
hey! and we even already have such a thread local available ❤️

Reply:
🙈